OctoConsulting/keip
# Support automatic reload of the Keip application when property source configmaps are updated

# Description
This PR will allow updates to property sources through Kustomize to automatically update the Spring Cloud K8s Config environment variable of a Keip contianer with the latest hashed configmap name for a given property source. The container will then be restarted per the Metacontroller's reconciliation process.

**Caveat**: The Metacontroller does not manage the lifecycle for `related_resources` like the Configmaps in this PR. Unused Configmaps will need to be removed manually or through some other mechanism.

# Testing
1. Build the lambda controller image under `operator/controller/webhook` with `make build`
2. Push the image to your k8s cluster
3. Deploy the metacontroller and keip operator with `make all`
4. Once the operator is up, deploy the example `kubectl apply -k example`
5. Tail the logs of the example container, update the application.yaml `greeting` variable, and re-apply `example`.
6. The container should restart and you should see the new message in the logs.
